numba/np/arrayobj.py

```diff
@@ -3026,6 +3026,14 @@ def array_is_impl(a, b):
 
     return context.compile_internal(builder, array_is_impl, sig, args)
 
+# ------------------------------------------------------------------------------
+# Hash
+
+
+@overload_method(types.Array, "__hash__")
+def ol_array_hash(arr):
+    return lambda arr: None
```

Comment:
Why not raise?

Reply:
The intention is to replicate the behaviour in Python, the `__hash__` attribute of unhashable types is often `None`, e.g. `np.ndarray`:
```pycon
In [1]: import numpy as np

In [2]: x = np.ones(3)

In [3]: print(x.__hash__)
None
```